tests/garage/tf/regressors/test_gaussian_cnn_regressor_with_model.py

```diff
@@ -0,0 +1,280 @@
+        assert np.array_equal(x_mean, x_mean_expected)
+        assert np.array_equal(x_std, x_std_expected)
+
+        y_mean = self.sess.run(gcr.model.networks['default'].y_mean)
+        y_mean_expected = np.zeros_like(y_mean)
+        y_std = self.sess.run(gcr.model.networks['default'].y_std)
+        y_std_expected = np.ones_like(y_std)
+
+        assert np.allclose(y_mean, y_mean_expected)
+        assert np.allclose(y_std, y_std_expected)
+
+    @pytest.mark.large
+    def test_fit_smaller_subsample_factor(self):
+        gcr = GaussianCNNRegressorWithModel(
+            input_shape=(10, 10, 3),
+            num_filters=(3, 6),
+            filter_dims=(3, 3),
+            strides=(1, 1),
+            padding='SAME',
+            hidden_sizes=(32, ),
+            output_dim=1,
+            subsample_factor=0.9,
+            adaptive_std=False)
+        train_data, test_data = get_train_test_data()
+        observations, returns = train_data
+
+        for _ in range(20):
+            gcr.fit(observations, returns)
+
+        paths, expected = test_data
+
+        prediction = gcr.predict(paths['observations'])
+        average_error = 0.0
+        for i in range(len(expected)):
+            average_error += np.abs(expected[i] - prediction[i])
+        average_error /= len(expected)
+        assert average_error <= 0.05
+
+    @pytest.mark.large
+    def test_fit_without_trusted_region(self):
+        gcr = GaussianCNNRegressorWithModel(
+            input_shape=(10, 10, 3),
+            num_filters=(3, 6),
+            filter_dims=(3, 3),
+            strides=(1, 1),
+            padding='SAME',
+            hidden_sizes=(32, ),
+            output_dim=1,
+            adaptive_std=False,
+            use_trust_region=False)
+        train_data, test_data = get_train_test_data()
+        observations, returns = train_data
+
+        for _ in range(20):
+            gcr.fit(observations, returns)
+
+        paths, expected = test_data
+
+        prediction = gcr.predict(paths['observations'])
+        average_error = 0.0
+        for i in range(len(expected)):
+            average_error += np.abs(expected[i] - prediction[i])
+        average_error /= len(expected)
+        assert average_error <= 0.05
+
+    @pytest.mark.parametrize('output_dim', [(1), (2), (3)])
+    def test_log_likelihood_sym(self, output_dim):
+        input_shape = (28, 28, 3)
+        gcr = GaussianCNNRegressorWithModel(
+            input_shape=input_shape,
+            num_filters=(3, 6),
+            filter_dims=(3, 3),
+            strides=(1, 1),
+            padding='SAME',
+            hidden_sizes=(32, ),
+            output_dim=1,
+            adaptive_std=False,
+            use_trust_region=False)
+
+        new_input_var = tf.placeholder(
+            tf.float32, shape=(None, ) + input_shape)
+        new_ys_var = tf.placeholder(
+            dtype=tf.float32, name='ys', shape=(None, output_dim))
+
+        data = np.random.random(size=input_shape)
```

Comment:
were you not able to find a way to test without random data?

Reply:
I didn't realize I was using random data in this test case, I'll fix it.